pubsub/google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -44,18 +44,25 @@ class Client(object):
     Args:
         batch_settings (~google.cloud.pubsub_v1.types.BatchSettings): The
             settings for batch publishing.
-        batch_class (class): A class that describes how to handle
+        batch_class (Optional[type]): A class that describes how to handle
             batches. You may subclass the
             :class:`.pubsub_v1.publisher.batch.base.BaseBatch` class in
             order to define your own batcher. This is primarily provided to
             allow use of different concurrency models; the default
             is based on :class:`threading.Thread`.
+        batch_lock (Optional[Any]): A lock used to ensure atomic modifications
```

Comment:
Hmm. I might prefer an interface where `batch_class` defined its lock class. Thoughts?

Reply:
I started out that way, but `threading.Lock` **is not** a class. Then I transitioned to a `Callable[[], Any]`. But having a class or callable is overkill: there is only one lock needed and it gets created when the object is constructed.

Reply:
Hmm. I am unconvinced, but I do not feel strongly about it either. :-)

Reply:
I misread Luke's initial comment. It turns out I agree with him.